frontend/src/test/test-helpers.js

```diff
@@ -1,21 +1,25 @@
 import { configure, mount } from 'enzyme';
+import { act } from 'react-dom/test-utils';
 import Adapter from 'enzyme-adapter-react-16';
 
 configure({ adapter: new Adapter() });
 
 let wrapper;
-export const mountToDoc = (reactElm) => {
+export const mountToDoc = async (reactElm) => {
   if (!document) {
     // Set up a basic DOM
-    global.document = jsdom('<!doctype html><html><body></body></html>');
+    global.document = jsdom('<!doctype html><html dir="ltr"><body></body></html>');
   }
   if (!wrapper) {
     wrapper = document.createElement('div');
     wrapper.id = 'app';
     document.body.appendChild(wrapper);
   }
-  const container = mount(reactElm);
-  // container.update();
+  let container;
+  await act(async () => {
```

Comment:
I'm curious why we're using the async version of `act()`. To a react novice (me), this looks a bit strange since the docs mostly just cover the synchronous version. I'm assuming it's because the react component may have called `useEffect()` and you want to be sure those callbacks have been completely processed before proceeding with the tests?

Reply:
This is one of the things I still don't enjoy/get about test frameworks in javascript -- it always seems more complicated to do "real" testing, or to test using the patterns/thoughts I have about testing, than I can find in the docs. :) For syllabus explorer we've been using `react-testing-library` as it is the default setup with create-react-app, which recommends using its own utility, `wait()`, to handle react component lifecycles and async fetches (see [How do I fix "an update was not wrapped in act(...)" warnings?](https://testing-library.com/docs/react-testing-library/faq)).

Reply:
Yes Artie, that is essentially correct. We want to wait for the lifecycle hooks to complete before the tests run. This approach is a bit of a Frankestein's Monster as I was attempting to work my way through Enzyme's `mount` issues with `useEffect`, so I wouldn't be surprised if there is a better approach than this.

The reason I went with Enzyme - to Elliott's point - is (a) because it is what I'm used to, and (b) because it can provided a decent output to pass to axe.